src/network_engine.cpp

```diff
@@ -666,9 +667,6 @@ NetworkEngine::send(const char *buf, size_t len, int flags, const SockAddr& addr
     if (sendto(s, buf, len, flags, addr.get(), addr.getLength()) == -1) {
         int err = errno;
         DHT_LOG.e("Can't send message to %s: %s", addr.toString().c_str(), strerror(err));
-        if (err == EPIPE) {
-            throw SocketException(EPIPE);
```

Comment:
EPIPE on UDP sockets is encountered on iOS and macOS devices when getting back from sleep. In this case the socket needs to be recreated. Since the socket is opened and provided by upper layers (DhtRunner) the error has to be propagated back to the DhtRunner,

Reply:
Thanks for your explanation. It’s a great help for me to understand how it is working. Thanks!